### ISSUE TITLE
Can't run tests if the package.json script is not named 'test'

### ISSUE THREAD

**Issue by roNn23** [ionic-team/stencil#2790]
**Stencil version:**
```
 @stencil/core@2.3.0
```

**I'm submitting a:**
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**
Create a script in `package.json` with the name `coverage` and the script content: `stencil test --spec --coverage`. Unfortunately it can't run, because Stencil reads the name of the script itself as argument for the CLI (`--coverage coverage`):
```
yarn coverage
yarn run v1.22.4
$ stencil test --spec --coverage
[34:43.8]  @stencil/core
[34:44.1]  v2.3.0 ⛵️
[34:44.4]  testing spec files
[34:44.5]  jest args: --coverage coverage --spec --max-workers=8
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In xxx
  97 files checked.
  testMatch:  - 0 matches
  testPathIgnorePatterns: xxx - 18 matches
  testRegex: (/__tests__/.*|\.?(test|spec|e2e))\.(tsx?|ts?|jsx?|js?)$ - 4 matches
Pattern: coverage - 0 matches
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Expected behavior:**
I can name my `package.json` script, however I want and the script runs anyway.

**Steps to reproduce:**
Create a script in `package.json` named other than 'test' and run it.

**Related code:**
```js
// package.json
{
  // ...
  "scripts": {
    "test": "stencil test --spec --coverage ", // runs
    "coverage": "stencil test --spec --coverage", // doesn't run
  },
  // ...
}
```

**Other information:**
The error probably occurs in the following file:
https://github.com/ionic-team/stencil/blob/cfd5d39bad4ce2cab431595ae8c8f11d1fabb192/src/testing/jest/jest-config.ts

It's also not possible to run other test scripts like: `"test:watch": "stencil test --spec --watchAll"`

**Comment by claviska** [comment]
I tried replicating this by changing the default `test` script to `coverage` but my tests seem to be working fine, both E2E and unit tests. Also tried with the `--coverage` flag just to be sure, but they ran as normal.

A few questions to help narrow this down:

- Can you verify that this happens with a starter app so we know it's nothing specific to your app?
- Does this happen if you use npm (curious if it has something to do with yarn passing the args)
- What version of Node are you using?
Thanks!

**Comment by claviska** [comment]
Could be related: https://github.com/ionic-team/stencil/issues/1609#issuecomment-755362126

> yarn requires you to use the run command (ie: yarn run test:foo), without run it won't work.

Can you try `yarn run coverage` to see if that works?

**Comment by roNn23** [comment]
Thanks for the quick reply!
It seems you're right, with `yarn run coverage` everything works, but without `run` it's not working. Tested it with the starter app, and there is the same behaviour. 

Well, the [yarn docs](https://classic.yarnpkg.com/en/docs/cli/run/) tells me that omitting `run` is valid, but it can be problematic:
> Note that built-in cli commands will have preference over your scripts, so you shouldn’t always rely on this shortcut in other scripts

But this is the first time that I came to this problem in 2 years. I'm working with the shortcuts the whole time. So I don't know if you want to fix this, I think you don't have to, but it would be nice 😎
